src/dbnode/storage/index/convert/convert_test.go

```diff
@@ -171,6 +172,113 @@ func TestFromSeriesIDAndTagIterReuseBytesFromSeriesId(t *testing.T) {
 	}
 }
 
+func TestFromSeriesIDAndEncodedTagsInvalid(t *testing.T) {
+	var (
+		id   = ident.StringID("foo")
+		tags = ident.NewTags(
+			ident.StringTag(string(convert.ReservedFieldNameID), "value"),
+		)
+		encodedTags = toEncodedTags(t, tags)
+	)
+	_, err := convert.FromSeriesIDAndEncodedTags(id, encodedTags)
+	assert.Error(t, err)
+}
+
+func TestFromSeriesIDAndEncodedTagsValid(t *testing.T) {
+	var (
+		id   = ident.StringID("foo")
+		tags = ident.NewTags(
+			ident.StringTag("bar", "baz"),
+		)
+		encodedTags = toEncodedTags(t, tags)
+	)
+	d, err := convert.FromSeriesIDAndEncodedTags(id, encodedTags)
+	assert.NoError(t, err)
+	assertContentsMatch(t, id, tags.Values(), d)
+}
+
+func TestFromSeriesIDAndEncodedTagsReuseBytesFromSeriesId(t *testing.T) {
```

Comment:
I think these two could be just a single test function for a happy path. `TestFromSeriesIDAndEncodedTagsReuseBytesFromSeriesId` verifies the logics in multiple cases, its coverage goes far beyond just byte reuse.

Reply:
https://github.com/m3db/m3/pull/3087/commits/a5436e8235c627486c9b0aec6ae438a82e5c30c7#diff-4b7b04cd8028a198a21da071cd0a8bbcc9d2f7fd0f6f4bd896a73019eccdfdfeL175-R183